### PR TITLE
Add new workspace connection types.

### PR DESCRIPTION
**Pull request by MilesHolland**
# Description

Adds support for 3 new workspace connection types: openAI, cognitive search, and cognitive services. This also removed support for a deprecated feature store connection type that is no longer included in the API version that adds the new WC types.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/ml/azure-ai-ml/CHANGELOG.md [hunk 1]
```diff
@@ -27,7 +29,7 @@
 
 - Improved the output when printing a workspace object to be more clean and readable.
 - Log level of unknown field notifications for pipeline nodes raised from INFO to WARNING.
-
+- Removed references to deprecated "feature_store" workspace connection type.
```
Comment (needuv) [inline sdk/ml/azure-ai-ml/CHANGELOG.md:32]: this should be in the 1.10.0 section

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_credentials.py [hunk 2]
```diff
@@ -776,3 +781,38 @@ def __eq__(self, other: object) -> bool:
         if not isinstance(other, AccessKeyConfiguration):
             return NotImplemented
         return self.access_key_id == other.access_key_id and self.secret_access_key == other.secret_access_key
+
+
+class ApiKeyConfiguration(RestTranslatableMixin, DictMixin):
```
Comment (needuv) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_credentials.py:786]: does this need to be experimental?

**Review by needuv — DISMISSED** [review]

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_credentials.py (hunk 2 above)
Comment (MilesHolland) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_credentials.py:787]: Good point, probably. Changed.

File: sdk/ml/azure-ai-ml/CHANGELOG.md (hunk 1 above)
Comment (MilesHolland) [reply sdk/ml/azure-ai-ml/CHANGELOG.md]: Moved.

**Review by needuv — APPROVED** [review]

**Pull request merged**